lib/kitchen/driver/ssh_base.rb

```diff
@@ -115,6 +115,14 @@ def verify(state) # rubocop:disable Metrics/AbcSize
       rescue Kitchen::Transport::TransportFailed => ex
         raise ActionFailed, ex.message
       ensure
+        if verifier[:scp_file]
```

Comment:
I'd love tho have this feature merged, looks like the the discussion about how to solve the rescue thing is not going anywhere. I do understand the concerns there is no an easy way to solve this issue.

Maybe to add as much as detail to the ActionFailed and SshFailed exceptions as possible to help the user understand the reason why the process failed is the only solution without change the ssh transport design.

IMHO this feature shouldn't be blocked any more time by this issue, I think this is super important to help identify errors in executions of more tan 3 or 4 tests suites in parallel. With the current implementation try to find the error in the log when you're running 15 or 16 parallel suites is a real pain.

👍